exonum/src/runtime/mod.rs

```diff
@@ -438,6 +438,30 @@ impl<T: Runtime> From<T> for Box<dyn Runtime> {
     }
 }
 
+/// Specifies system identifier for a [`Runtime`].
+pub trait WellKnownRuntime: Runtime {
+    /// Identifier of this runtime.
+    const ID: u32;
+}
+
+#[derive(Debug)]
+/// Instance of [`Runtime`] with corresponding ID.
+pub struct RuntimeInstance {
```

Comment:
Not sure that exposing `RuntimeInstance` APIs publicly is optimal (or necessary); I'd suggest to investigate this later.